src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -872,7 +877,18 @@ public function getModType(Expr $left, Expr $right, callable $getTypeCallback):
 			}
 		}
 
+		$float = new FloatType();
+		if ($leftType instanceof UnionType && $leftType->isSuperTypeOf($float)->yes()) {
+			return new ErrorType();
```

Comment:
How the type system responds shouldn't change because it's not actually true. The `%` operator still returns a number but only on some PHP versions it throws a deprecation notice.